tests/objc-cli/Makefile

```diff
@@ -52,3 +52,13 @@ test-xctest-leaks: ../leaktest/bin/Debug/leaktest.exe $(MANAGED_DLL) libmanaged.
 
 libLeakCheckAtExit.dylib: leak-at-exit.c
 	clang -arch i386 -arch x86_64 -shared $< -o $@
+
+test-static: test-static-macos test-static-ios test-static-tvos test-static-watchos
+
+test-static-%: managed.dll $(OBJC_GEN)
+	/Library/Frameworks/Mono.framework/Versions/Current/Commands/mono --debug $(OBJC_GEN) --debug managed.dll -c --outdir=build/$@-temp-dir --target=staticlibrary --platform=$*
+
+test-dynamic: test-dynamic-macos test-dynamic-ios test-dynamic-tvos test-dynamic-watchos
```

Comment:
maybe `dylib` and `framework` when that distinction is important (in this case it is, but not yet), since both `shared` and `dynamic` are technically correct for both of them.

And then also choose between `shared` and `dynamic` of course.